openmmtools/alchemy.py

```diff
@@ -1415,397 +1701,114 @@ def _alchemicallyModifyGBSAOBCForce(self, system, reference_force, force_labels,
             # Retrieve parameters.
             [charge, radius, scaling_factor] = reference_force.getParticleParameters(particle_index)
             # Set particle parameters.
-            if particle_index in self.ligand_atoms:
+            if particle_index in alchemical_region.alchemical_atoms:
                 parameters = [charge, radius, scaling_factor, 1.0]
             else:
                 parameters = [charge, radius, scaling_factor, 0.0]
             custom_force.addParticle(parameters)
 
-        # Add alchemically-modified GBSAOBCForce to system.
-        force_index = system.addForce(custom_force)
-        force_labels['alchemically modified GBSAOBCForce'] = force_index
+        return [custom_force]
 
-    def _createAlchemicallyModifiedSystem(self, mm=None):
-        """
-        Create an alchemically modified version of the reference system with global parameters encoding alchemical parameters.
+    # -------------------------------------------------------------------------
+    # Internal usage: Infer force labels
+    # -------------------------------------------------------------------------
 
-        Returns
-        -------
-        system : simtk.openmm.System
-            Alchemically-modified version of self.reference_system
-        force_labels : dict of int : str
-            force_labels[name] is the force index in the alchemically modified system of the modified force `name`
+    @staticmethod
+    def _find_force_components(alchemical_system):
+        """Return force labels and indices for each force."""
+        def add_label(label, index):
```

Comment:
Does this even work with `force_labels` targeting something in its outer scope that has not yet been defined? For at least clarity, could you move the variable initialization above the internal functions?